app/models/license.rb

```diff
@@ -14,9 +15,12 @@ class License < ApplicationRecord
   belongs_to :policy
   belongs_to :group,
     optional: true
+  belongs_to :user,
+    optional: true
   has_one :product, through: :policy
   has_many :license_users, validate: false, index_errors: true, dependent: :destroy_async
-  has_many :users, index_errors: true, through: :license_users
+  has_many :licensees, index_errors: true, through: :license_users, source: :user
+  union_of :users, sources: %i[licensees user], class_name: User.name
```

Comment:
Using `class_name:` like this could result in load errors if dependencies are cyclic. Use a string instead.